src/App.tsx

```diff
@@ -1,61 +1,136 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import { FormEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todosWithUsers = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosWithUsers);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasError, setHasError] = useState({
+    title: false,
+    userId: false,
+  });
+
+  const maxId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const handleTitle = (event: FormEvent) => {
+    const target = event.target as HTMLInputElement;
+
+    setTitle(target.value);
+    setHasError(prev => ({
+      ...prev,
+      title: false,
+    }));
+  };
```

Comment:
[CODE STYLE] - Don't use one-letter variable naming. It could be confusing for another developer, who reads your code. Also, it would really hard to find this variable in the code editor using search. Instead of `(event: FormEvent)`, you could use `(titleChangeEvent: FormEvent)`